pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -153,3 +189,103 @@ func createVeleroBackupObject(client client.Client, namespace string) error {
 
 	return client.Create(context.TODO(), backup)
 }
+
+func (r *ReconcileBackup) getAllHiveObjectsInNamespace(namespace string) ([]*hiveObject, error) {
+	nsLogger := addNamespaceLoggerFields(r.logger, namespace)
+
+	allNSObjs := []runtime.Object{}
+
+	allClusterDeployments := &hivev1.ClusterDeploymentList{}
+	err := r.List(context.TODO(), allClusterDeployments, client.InNamespace(namespace))
+	if err != nil {
+		nsLogger.WithError(err).Error("Couldn't get list of ClusterDeployments")
+		return nil, err
+	}
+
+	for i := range allClusterDeployments.Items {
+		allNSObjs = append(allNSObjs, &allClusterDeployments.Items[i])
+	}
+
+	allSyncSets := &hivev1.SyncSetList{}
+	err = r.List(context.TODO(), allSyncSets, client.InNamespace(namespace))
+	if err != nil {
+		nsLogger.WithError(err).Error("Couldn't get list of SyncSets")
+		return nil, err
+	}
+
+	for i := range allSyncSets.Items {
+		allNSObjs = append(allNSObjs, &allSyncSets.Items[i])
+	}
+
+	allHiveObjects := []*hiveObject{}
+	for _, nsObj := range allNSObjs {
+		hobj, err := newHiveObject(nsObj, r.checksumFunc, r.logger)
+		if err != nil {
+			return nil, err
+		}
+
+		allHiveObjects = append(allHiveObjects, hobj)
+	}
+
+	return allHiveObjects, nil
+}
+
+func (r *ReconcileBackup) getModifiedHiveObjectsInNamespace(namespace string) ([]*hiveObject, error) {
+	allHiveObjects, err := r.getAllHiveObjectsInNamespace(namespace)
```

Comment:
I don't see a compelling need to build up the slice of all hive objects. Let's process each hive object as we get it instead.